pubsublite-kafka-shim/src/main/java/com/google/cloud/pubsublite/kafka/PubsubLiteConsumer.java

```diff
@@ -0,0 +1,555 @@
+      Set<Partition> removed = new HashSet<>(previousAssignment);
+      removed.removeAll(newAssignment);
+      Set<Partition> added = new HashSet<>(newAssignment);
+      added.removeAll(previousAssignment);
+      if (!removed.isEmpty()) {
+        listener.onPartitionsLost(
+            removed.stream().map(this::toTopicPartition).collect(Collectors.toSet()));
+      }
+      if (!added.isEmpty()) {
+        listener.onPartitionsAssigned(
+            added.stream().map(this::toTopicPartition).collect(Collectors.toSet()));
+      }
+      consumer.get().setAssignment(newAssignment);
+      lastPartitions.set(newAssignment);
+    };
+  }
+
+  @Override
+  public void subscribe(Collection<String> collection) {
+    subscribe(collection, new NoOpRebalanceListener());
+  }
+
+  @Override
+  public void subscribe(
+      Collection<String> collection, ConsumerRebalanceListener consumerRebalanceListener) {
+    if (collection.size() != 1) {
+      throw new UnsupportedOperationException(
+          "Subscribing to multiple topics is not available for Pub/Sub Lite.");
+    }
+    checkTopic(collection.iterator().next());
+    if (consumer.isPresent()) {
+      if (assigner.isPresent()) {
+        // No-op: we can only subscribe to one topic and we already are.
+        return;
+      }
+      throw new IllegalStateException("Called subscribe after calling assign.");
+    }
+    consumer = Optional.of(consumerFactory.newConsumer());
+    try {
+      assigner = Optional.of(assignerFactory.New(newAssignmentReceiver(consumerRebalanceListener)));
+    } catch (StatusException e) {
+      throw toKafka(e);
+    }
+  }
+
+  @Override
+  public void assign(Collection<TopicPartition> collection) {
+    if (collection.isEmpty()) {
+      unsubscribe();
+      return;
+    }
+    if (assigner.isPresent()) {
+      throw new IllegalStateException("Called assign after calling subscribe.");
+    }
+    Set<Partition> partitions =
+        collection.stream().map(this::checkTopicGetPartition).collect(Collectors.toSet());
+    if (!consumer.isPresent()) {
+      consumer = Optional.of(consumerFactory.newConsumer());
+    }
+    consumer.get().setAssignment(partitions);
+  }
+
+  @Override
+  public void unsubscribe() {
+    assigner.ifPresent(instance -> instance.stopAsync().awaitTerminated());
+    assigner = Optional.empty();
+    consumer.ifPresent(instance -> instance.close(INFINITE_DURATION));
+    consumer = Optional.empty();
+  }
+
+  private static Duration toDuration(long l, TimeUnit timeUnit) {
+    return Duration.ofMillis(TimeUnit.MILLISECONDS.convert(l, timeUnit));
+  }
+
+  @Override
+  public ConsumerRecords<byte[], byte[]> poll(long l) {
+    return poll(Duration.ofMillis(l));
+  }
+
+  @Override
+  public ConsumerRecords<byte[], byte[]> poll(Duration timeout) {
+    return requireValidConsumer().poll(timeout);
+  }
+
+  Map<Partition, Offset> checkAndTransform(Map<TopicPartition, OffsetAndMetadata> map) {
```

Comment:
I would personally make these method names more descriptive. "checkAndTransform" doesn't really convey what this method is doing without looking at the return/argument types, which are not always immediately visible at the call sites.

There's others like this: toKafka, toDuration, etc, but I think those are more obvious.

Reply:
Done.